core/state_processor_svm.go

```diff
@@ -0,0 +1,326 @@
+	if msg.To() == nil {
+		receipt.ContractAddress = crypto.CreateAddress(msg.From(), tx.Nonce())
+	}
+
+	// Set the receipt logs and create a bloom for filtering
+	receipt.Logs = statedb.GetLogs(tx.Hash())
+	receipt.Bloom = types.CreateBloom(types.Receipts{receipt})
+
+	// Free the patch and destroy machine
+	vm.Free()
+	patch.Free()
+
+	return receipt, gas, err
+}
+
+func makeSputnikVMPatch(config *params.ChainConfig, header *types.Header) sputnikvm.DynamicPatch {
+	gasTable := config.GasTable(header.Number)
+
+	// Zero == unlimited
+	codeDepositLimit := 0
+	if config.IsEIP170F(header.Number) {
+		codeDepositLimit = params.MaxCodeSize
+	}
+
+	// Helper to convert uint64 tp big.Int
+	toBigInt := func(x uint64) *big.Int {
+		return new(big.Int).SetUint64(x)
+	}
+
+	rules := config.Rules(header.Number)
+
+	// Calculate the upfront CREATE cost (it's lower on Frontier)
+	createGasCost := params.CreateGas
+	if !rules.IsEIP2F {
+		createGasCost = 0
+	}
+
+	// Build list of enabled precompile
+	enabledPrecompileds := [][20]byte{
+		common.BytesToAddress([]byte{1}),
+		common.BytesToAddress([]byte{2}),
+		common.BytesToAddress([]byte{3}),
+		common.BytesToAddress([]byte{4}),
+	}
+
+	if rules.IsEIP198F {
+		enabledPrecompileds = append(enabledPrecompileds,
+			common.BytesToAddress([]byte{5}))
+	}
+
+	if rules.IsEIP213F {
+		enabledPrecompileds = append(enabledPrecompileds,
+			common.BytesToAddress([]byte{6}))
+		enabledPrecompileds = append(enabledPrecompileds,
+			common.BytesToAddress([]byte{7}))
+	}
+
+	if rules.IsEIP212F {
+		enabledPrecompileds = append(enabledPrecompileds,
+			common.BytesToAddress([]byte{8}))
+	}
+
+	patchBuilder := sputnikvm.DynamicPatchBuilder{
+		CodeDepositLimit:            uint(codeDepositLimit),
+		CallStackLimit:              uint(params.CallCreateDepth),
+		GasExtcode:                  toBigInt(gasTable.ExtcodeCopy),
+		GasBalance:                  toBigInt(gasTable.Balance),
+		GasSload:                    toBigInt(gasTable.SLoad),
+		GasSuicide:                  toBigInt(gasTable.Suicide),
+		GasSuicideNewAccount:        toBigInt(gasTable.CreateBySuicide),
+		GasCall:                     toBigInt(gasTable.Calls),
+		GasExpbyte:                  toBigInt(gasTable.ExpByte),
+		GasTransactionCreate:        toBigInt(createGasCost),
+		ForceCodeDeposit:            !rules.IsEIP2F,
+		HasDelegateCall:             rules.IsEIP7F,
+		HasStaticCall:               rules.IsEIP214F,
+		HasRevert:                   rules.IsEIP140F,
+		HasReturnData:               rules.IsEIP211F,
+		HasBitwiseShift:             rules.IsEIP145F,
+		HasCreate2:                  rules.IsEIP1014F,
+		HasExtCodeHash:              rules.IsEIP1052F,
+		HasReducedSstoreGasMetering: rules.IsEIP1283F,
+		ErrOnCallWithMoreGas:        !rules.IsEIP150,
+		CallCreateL64AfterGas:       rules.IsEIP150,
+		MemoryLimit:                 ^uint(0), // Reversed 0 is max unsigned integer value for uint
```

Comment:
:white_flower:

Reply:
@meowsbits afaik I couldn’t find any parameter that defines memory limit, and the classic geth w/ SVM was operating the same way, with an unlimited memory storage. Though that’s pretty much welcoming for a DOS attack :)

Is there any limit definition I could use?

Reply:
https://github.com/multi-geth/multi-geth/blob/master/core/vm/memory_table.go
https://github.com/multi-geth/multi-geth/blob/master/core/vm/common.go#L38

[18446744073709551615](https://docs.microsoft.com/en-us/dotnet/api/system.uint64.maxvalue?view=netframework-4.8)

```go
$ cat max-uint64.go 
package main

import (
        "fmt"
        "log"
        "math/big"
)

func main() {
        maybeMax := new(big.Int).SetUint64(18446744073709551615)
        b := new(big.Int).Sub(maybeMax, big.NewInt(10))
        for {
                b = b.Add(b, big.NewInt(1))
                if !b.IsUint64() {
                        log.Fatal("max uint64 is", b, "-1")
                } else {
                        fmt.Println(b)
                }
        }
}

```